docs/research/USB_compliance_test.md

```diff
@@ -0,0 +1,212 @@
+- analyze LFPS at communication initialization
+- analyze Training Sequences TSEQ, TS1, TS2 Ordered Sets
+- analyze State Machine (check if states proceed according to Specification)
+- checking if USB Port/Device ends in U0 mode
+- checking packets between host and recognized device
+
+>USB-IF has its own Link Layer Compliance Tests program. However, to carry out
+those tests it is required to use their hardware and software tools. In my
+opinion, good enough alternative is to use one of mentioned above devices,
+because they have features which are sufficient to resolve stick detection
+problems.
+
+If mentioned above analyses aren't sufficient, tests from sections 7.5.1 to
+7.5.5 from [USB 3.1 Link Layer Test Specification](https://www.usb.org/sites/default/files/USB_3_1_Link_Layer_Test_Specification_2018_01_02.pdf)
+can be performed.
+
+
+## Additional conclusions
+
+#### USB 3.0 analyze
+
+Mentioned above equipment is necessary to carry out electrical and link layer
+tests. Due to its high price it's rather impossible to perform such tests now.
+So far I couldn't find any alternate solutions which will meet the requirements
+and will be low-price devices. It results from the fact, that to test USB 3.0
+protocol, high frequency devices are needed and such devices are rather
+expensive. Most of available instruments are suitable only for USB 2.0 testing.
+
+#### USB 2.0 analyze
+
+USB 2.0 Analyzer's principles of operation is exactly the same as USB 3.0
+Analyzer and it is described above in that document.
+
+Proposed USB 2.0 Analyzers with dedicated software:
+- OpenVizsla | Wireshark
+- Beagle USB 480 | Data Center Software
+
+Both analyzers are perfectly good to capture and investigate USB data. However,
+OpenVizsla is an open-source solution which can easily handle with USB analysis
+within low price. In combination with Wireshark (or any other program which can
+descramble USB data) it doesn't stand out from any other available solutions. On
+the other hand, Beagle's software has feature which analyze state machine and
+show current state in LTSSM.
+
+>Those devices can't be extended to analyze USB 3.0/3.1 transmission.
+
+#### RxDetect state
+
+Detailed description of `RxDetect` machine state (with substates) is available
+in Section 7.5.3 in USB 3.0 Specification.
+
+Initially USB ports are in `SS.Disabled` state (SuperSpeed connection is cut
+off). `RxDetect` state (in fact `RxDetect.Reset`) is reached after Power On
+Reset or Directed Warm Reset.
+
+Let's look how machine substate sequence looks like depending on scenarios:
+
+- RxDetect.Reset
+
+  If it was reached by power on reset it transit directly to `RxDetect.Active`.
+
+  If it was reached by warm reset it transmits LFPS sequence (warm reset
+  sequence) and goes to `RxDetect.Active`.
+
+- RxDetect.Active
+
+  In this state receiver termination is performed. If it is detected then
+  `Polling.LFPS` state is reached. If it is note detected then `RxDetect.Quiet`
```

Comment:
What is the timeout of reaching `Polling.LFPS` after which the `RxDetect.Quiet` kicks in?

Reply:
Receiver termination means detecting termination at line. If it is not detected then it goes to RxDetect.Quiet state (after 12ms goes back and try again), if it is detected (it means there is usb 3.0 device at the end) it goes to Polling.LFPS state. There is no timeout of reaching Polling.LFPS then. I can rewrite it, so it will be understandable